test/addons/openssl-test-engine/test.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+const common = require('../../common');
+
+// This tests crypto.setEngine().
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+const engine = path.join(__dirname,
+                         `/build/${common.buildType}/testsetengine.engine`);
+
+if (!fs.existsSync(engine))
+  common.skip('no engine');
+
+assert.throws(() => crypto.setEngine(true), /ERR_INVALID_ARG_TYPE/);
+assert.throws(() => crypto.setEngine('/path/to/engine', 'notANumber'),
+              /ERR_INVALID_ARG_TYPE/);
+
+{
+  const invalidEngineName = 'xxx';
+  assert.throws(() => crypto.setEngine(invalidEngineName),
+                /ERR_CRYPTO_ENGINE_UNKNOWN/);
+  assert.throws(() => crypto.setEngine(invalidEngineName,
+                                       crypto.constants.ENGINE_METHOD_RSA),
+                /ERR_CRYPTO_ENGINE_UNKNOWN/);
+}
+
+crypto.setEngine('dynamic');
+crypto.setEngine('dynamic');
+
+crypto.setEngine('dynamic', crypto.constants.ENGINE_METHOD_RSA);
+crypto.setEngine('dynamic', crypto.constants.ENGINE_METHOD_RSA);
+
```

Comment:
```suggestion

const engine = path.join(__dirname,
                         `/build/${common.buildType}/testsetengine.engine`);

if (!fs.existsSync(engine))
  common.skip('no engine');

```
and let's place it here